### ISSUE TITLE
FYI

### ISSUE THREAD

**Issue by Thorin-Oakenpants** [sgmenda/browser-fingerprinting#1]
I had a chat (on github) with @pes10k about this, that there are only so many timezone aliases - you can use this [PoC](https://arkenfox.github.io/TZP/tests/timezones.html) - I can add you to that private repo if you like, just let me know

I designed it to see how many unique results I could get using timezone offsets, and I'm using date.parse() and epoch time as it's very difficult for extensions to tamper with, not the PoC, but the actual test on tzp, As a side effect, it groups aliased or identical timezones together

Now the minimum test for max results is 5 years (2 dates per year). This gives 374 timezone groups and means all dates, calendars etc would not be "broken" (i.e unintended results for the end user) for any date ever.

I confused myself in the chat, it had been quite a while since I built the PoC, but essentially, if you were to only use years going forward, say from 2018, just use `2018, 2019, 2020, 2021` in the text field (so some calendar backward compat), then you could get away with as few 70 timezones, of which 24 will be unique (i.e no randomizing possible, and those are likely to always be a problem), but other `46` groups covering the remaining `431` timezones allow randomizing with no "breakage" - i.e, as long as it is made clear in documentation that it is forward compatible only.

Note: some of those groups are small (2 or 3), but I doubt you can do anything about them. And some can be misleading, for example 2018+ `Antarctica/McMurdo, Pacific/Auckland, Pacific/Fiji` .. or my 5 year PoC example: `Antarctica/McMurdo, Pacific/Auckland` ... who is at McMurdo, might as well just say you're New Zealand time?

edit: this was meant to be about your timezone comments at https://github.com/sgmenda/ideas

**Comment by initsecret** [comment]
Thank you for the detailed response and for collecting the data.

I assume this is with reference to [my (silly) comment](https://github.com/brave/brave-browser/issues/8574#issuecomment-848699474) on the brave issue "Fingerprinting 3.0: Simplified timezones". Unfortunately, I haven't thought too much about fingerprinting timezones besides that comment. 😕

I just [commented](https://bugzilla.mozilla.org/show_bug.cgi?id=1709867#c1) on the make-RFP-timezone-spoofing-finegrained bug to clarify my thinking some more.

I maintain that simplifying timezones is a super cool idea but I am not sure if I have the bandwidth to work on it right now (cuz of the math stuff and the canvas stuff we are working on). Could you post a link to this comment on the [make-RFP-timezone-spoofing-finegrained bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1709867) so it'll be easy for when people get to this bug. (I don't think anyone except me looks at this repo, I created it a few days ago to host a chromium webaudio test to do some experiments for the math stuff we are working on. 🙂)

update: hehe, I didn't see the edit. I reckon my guess was kinda right. 🙂

**Comment by Thorin-II** [comment]
> Could you post a link to this comment

no need, it's already linked by virtue of you hyperlinking to it, and Peter is already aware

> I just commented on the make-RFP-timezone-spoofing-finegrained bug to clarify my thinking some more

I don't really agree on this. The only thing this achieves is better "compat" (it's not breakage per se IMO), or rather expected user outcomes (which is not a bad thing), but at the expense of sacrificing a number of users into very small buckets - just on the math of it all, it's a terrible idea to introduce 60 or 70 buckets (despite the randomness, trivial to bypass to a common result) when we go to such lengths as to tighten a binary result like prefers-color-scheme.

The real answer is to educate users: and/or have an RFP slider (for Firefox users, not necessarily for Tor Browser), like Brave Shield's off, standard, and strict, and/or allow an exception per site like canvas for the most "irritating" items end-users seem to hate like timezone or even prefers-color: I don't see allowing the real timezone as an issue for users who allow it on select sites, especially if they're logged in, e.g. gmail: it's not like they would be revealing it across all sites.

We should be very careful after all this time allowing relaxing of certain metrics or introducing system wide toggles

**Comment by initsecret** [comment]
Ah, good catch, my comment was poorly worded, sorry about that, I meant make (2) the default in Firefox, not in RFP. For RFP, I was thinking of the following userflow:

1. user enables RFP or uses the Tor Browser
2. user visits `foo.bar`, it gets UTC
3. user exempts `foo.bar` by adding `foo.bar` to `privacy.resistFingerprinting.exemptedDomains`
4. user visits `foo.bar`, it gets UTC+CorrectOffset and simplified timezone name (which is possibly wrong)
5. if users wants exempted domains to get the correct timezone name, they can disable `privacy.resistfingerprinting.simplifyTimezones` or if they just want to exempt this domain they can add it to `privacy.resistfingerprinting.simplifyTimezonesExemptList`.

**Comment by Thorin-II** [comment]
I don't think it is worth it TBH: 60 or 70 buckets is boring ;) and breaking from at a minimum of 374-valid-and-all-different timezones for **all** users by default users would probably be a no-no. It would have to be behind a fingerprinting pref .. which we already have, and we use UTC0 :) no need to reinvent the wheel.

One thing I think we should do for **all** users is hardcode hardwareConcurrency, make outer = inner (there's a ticket for that), remove screenx/y (ticket for that too), give everyone scrollbar overlays, turn the findbar into an overlay, make available screen = screen, and mask screen on a sliding scale based on inner window - none of these break anything
- ^ I'd do all that first before timezones

We need to get RFP **front facing** with site exemptions and some **education** on the few things that break: e.g. canvas solution allow a site exception, shortcut keys on some sites, and timing issues with some anims etc. And also educate them on the timezone. And reduce some more barriers to uptake: inner window real estate is a big issue (and we can easily expand the width for desktops with no entropy gain - there's a ticket for this as well) .. and do something about webgl, but not as a roadblock.

We need to get tom to do dishes more often :)
